src/seedu/addressbook/data/person/ReadOnlyPerson.java

```diff
@@ -33,28 +33,44 @@ default boolean isSameStateAs(ReadOnlyPerson other) {
     }
 
     /**
+     * Returns a printableString
+     */
+    default String getPrintableString(Printable... printables) {
+        StringBuilder stringBuilder = new StringBuilder();
+        for (Printable p: printables) {
+            stringBuilder.append(p.getPrintableString()).append(" ");
+        }
+        return stringBuilder.toString();
+    }
+
+    /**
      * Formats the person as text, showing all contact details.
      */
     default String getAsTextShowAll() {
         final StringBuilder builder = new StringBuilder();
-        final String detailIsPrivate = "(private) ";
-        builder.append(getName())
-                .append(" Phone: ");
-        if (getPhone().isPrivate()) {
-            builder.append(detailIsPrivate);
-        }
-        builder.append(getPhone())
-                .append(" Email: ");
-        if (getEmail().isPrivate()) {
-            builder.append(detailIsPrivate);
-        }
-        builder.append(getEmail())
-                .append(" Address: ");
-        if (getAddress().isPrivate()) {
-            builder.append(detailIsPrivate);
-        }
-        builder.append(getAddress())
-                .append(" Tags: ");
+        String printableString = getPrintableString(getName(), getPhone(), getEmail(), getAddress());
+//        final String detailIsPrivate = "(private) ";
+//        builder.append(getName())
```

Comment:
Please don't leave around non-functional code commented! You can always delete and commit so that you can revert to a previous version.